src/util.c

```diff
@@ -0,0 +1,140 @@
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/wait.h>
+
+#include "util.h"
+#include "log.h"
+
+int email_body_size = 0;
```

Comment:
There is no need to keep this here. Since you give `email_body` as an argument to the functions, you can do it with this variable too.